shotover-proxy/src/frame/cassandra.rs

```diff
@@ -462,38 +515,512 @@ impl CassandraOperation {
+                if CQLStatement::has_params_in_relation_elements(&update.if_clause) {
+                    return true;
+                }
+            }
+            _ => {}
+        }
+        false
+    }
+}
+
+impl Display for CQLStatement {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        self.statement.fmt(f)
+    }
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct CQL {
+    pub statements: Vec<Box<CQLStatement>>,
+    pub(crate) has_error: bool,
+}
+
+impl CQL {
+    /// the number of statements in the CQL
+    pub fn get_statement_count(&self) -> usize {
+        self.statements.len()
+    }
+
+    fn from_value_and_col_spec(value: &Value, col_spec: &ColSpec) -> Operand {
+        match value {
+            Value::Some(vec) => {
+                let cbytes = CBytes::new(vec.clone());
+                let message_value =
+                    MessageValue::build_value_from_cstar_col_type(col_spec, &cbytes);
+                let pmsg_value = &message_value;
+                pmsg_value.into()
+            }
+            Value::Null => Operand::Null,
+            Value::NotSet => Operand::Null,
+        }
+    }
+    fn set_param_value_by_name(
+        name: &str,
+        query_params: &QueryParams,
+        param_types: &[ColSpec],
+    ) -> Operand {
+        if let Some(QueryValues::NamedValues(value_map)) = &query_params.values {
+            if let Some(value) = value_map.get(name) {
+                if let Some(idx) = value_map
+                    .iter()
+                    .enumerate()
+                    .filter_map(
+                        |(idx, (key, _value))| {
+                            if key.eq(name) {
+                                Some(idx)
+                            } else {
+                                None
+                            }
+                        },
+                    )
+                    .next()
+                {
+                    return CQL::from_value_and_col_spec(value, &param_types[idx]);
+                }
+            }
+        }
+        Operand::Param(format!(":{}", name))
+    }
+
+    fn set_param_value_by_position(
+        param_idx: &mut usize,
+        query_params: &QueryParams,
+        param_types: &[ColSpec],
+    ) -> Operand {
+        if let Some(QueryValues::SimpleValues(values)) = &query_params.values {
+            if let Some(value) = values.get(*param_idx) {
+                *param_idx += 1;
+                CQL::from_value_and_col_spec(value, &param_types[*param_idx])
+            } else {
+                *param_idx += 1;
+                Operand::Param("?".into())
+            }
+        } else {
+            *param_idx += 1;
+            Operand::Param("?".into())
+        }
+    }
+
+    fn set_operand_if_param(
+        operand: &Operand,
+        param_idx: &mut usize,
+        query_params: &QueryParams,
+        param_types: &[ColSpec],
+    ) -> Operand {
+        match operand {
+            Operand::Tuple(vec) => {
```

Comment:
With iterators: 

```rust
Operand::Tuple(vec) => Operand::Tuple(
    vec.iter()
        .map(|o| CQL::set_operand_if_param(o, param_idx, query_params, param_types))
         .collect(),
),
```

Also is it possible to pass a `&mut Operand` instead of creating a new one?